src/utils/managing.officer.corporate.ts

```diff
@@ -0,0 +1,140 @@
+import { NextFunction, Request, Response } from "express";
+import { Session } from "@companieshouse/node-session-handler";
+import { v4 as uuidv4 } from 'uuid';
+
+import { logger } from "./logger";
+import { saveAndContinue } from "./save.and.continue";
+import { ApplicationDataType } from "../model";
+import {
+  getFromApplicationData,
+  mapDataObjectToFields,
+  mapFieldsToDataObject,
+  prepareData,
+  removeFromApplicationData,
+  setApplicationData,
+} from "./application.data";
+import { ManagingOfficerCorporateKey, ManagingOfficerCorporateKeys } from "../model/managing.officer.corporate.model";
+import {
+  AddressKeys,
+  HasSamePrincipalAddressKey,
+  ID,
+  IsOnRegisterInCountryFormedInKey,
+  PublicRegisterNameKey,
+  RegistrationNumberKey
+} from "../model/data.types.model";
+import { PrincipalAddressKey, PrincipalAddressKeys, ServiceAddressKey, ServiceAddressKeys } from "../model/address.model";
+
+export const getManagingOfficerCorporate = (req: Request, res: Response, templateName: string, backLinkUrl: string) => {
+  logger.debugRequest(req, `${req.method} ${req.route.path}`);
+
+  return res.render(templateName, {
+    backLinkUrl: backLinkUrl,
+    templateName: templateName
+  });
+};
+
+export const getManagingOfficerCorporateById = (req: Request, res: Response, next: NextFunction, templateName: string, backLinkUrl: string) => {
+  try {
+    logger.debugRequest(req, `GET BY ID ${req.route.path}`);
+
+    const id = req.params[ID];
+    const data = getFromApplicationData(req, ManagingOfficerCorporateKey, id, true);
```

Comment:
I changed data to officerData. You might like to do the same, but you don't need to